### ISSUE TITLE
Return promises from service methods

### ISSUE THREAD

**Issue by sivakumar-kailasam** [PrinceCornNM/ember-algolia#4]
Having the service methods return promises instead of taking callbacks as inputs would probably make them more ember-friendly since most folks might end up wrapping the service calls in promises again.

**Comment by conradlz** [comment]
Yeah,  I find that they end up getting wrapped in promises in most use
cases anyway.  I'll explore ways to provide some meaningful promises which
could be more convenient. I'd also really appreciate some context for
similar suggested functionality.

**Comment by sivakumar-kailasam** [comment]
You can use [RSVP's denodeify](https://github.com/tildeio/rsvp.js/blob/master/lib/rsvp/node.js#L134) to convert from node style callbacks to promises. Here's a sampler: https://github.com/ember-learn/ember-api-docs/pull/233/files#diff-319f96f3a3558fc217c4e30a5fd10487R50

**Comment by bradhaydon** [comment]
Would you mind giving an example of how to use denodeify and ember-algolia? I could not get it to work properly. I really appreciate it!
